CRM/Price/Form/Preview.php

```diff
@@ -132,7 +132,9 @@ public function buildQuickForm() {
         foreach ($group['fields'] as $field) {
           $fieldId = $field['id'];
           $elementName = 'price_' . $fieldId;
-          CRM_Price_BAO_PriceField::addQuickFormElement($this, $elementName, $fieldId, FALSE, $field['is_required']);
+          if (!empty($field['options'])) {
+            CRM_Price_BAO_PriceField::addQuickFormElement($this, $elementName, $fieldId, FALSE, $field['is_required']);
+          }
```

Comment:
Due to this `Preview` action of a text field displays nothing. To replicate - Create a text price field and click on Preview. As a text field doesn't have any options, no element is being added.